include/mlir/Dialect/LLVMIR/LLVMOps.td

```diff
@@ -684,4 +684,114 @@ def LLVM_ExpOp : LLVM_Op<"intr.exp", [NoSideEffect]>,
     $res = builder.CreateCall(fn, {$in});
   }];
 }
+
+def LLVM_FAbsOp : LLVM_Op<"intr.fabs", [NoSideEffect]>,
+                   Arguments<(ins LLVM_Type:$in)>,
+                   Results<(outs LLVM_Type:$res)> {
+  let llvmBuilder = [{
+    llvm::Module *module = builder.GetInsertBlock()->getModule();
+    llvm::Function *fn = llvm::Intrinsic::getDeclaration(
+        module, llvm::Intrinsic::fabs, {$in->getType()});
+    $res = builder.CreateCall(fn, {$in});
+  }];
+}
+
+def LLVM_FCeilOp : LLVM_Op<"intr.ceil", [NoSideEffect]>,
+                   Arguments<(ins LLVM_Type:$in)>,
+                   Results<(outs LLVM_Type:$res)> {
+  let llvmBuilder = [{
+    llvm::Module *module = builder.GetInsertBlock()->getModule();
+    llvm::Function *fn = llvm::Intrinsic::getDeclaration(
+        module, llvm::Intrinsic::ceil, {$in->getType()});
+    $res = builder.CreateCall(fn, {$in});
+  }];
+}
+
+def LLVM_CosOp : LLVM_Op<"intr.cos", [NoSideEffect]>,
+                   Arguments<(ins LLVM_Type:$in)>,
+                   Results<(outs LLVM_Type:$res)> {
+  let llvmBuilder = [{
+    llvm::Module *module = builder.GetInsertBlock()->getModule();
+    llvm::Function *fn = llvm::Intrinsic::getDeclaration(
+        module, llvm::Intrinsic::cos, {$in->getType()});
+    $res = builder.CreateCall(fn, {$in});
+  }];
+}
+
+def LLVM_CopySignOp : LLVM_Op<"intr.copysign", [NoSideEffect]>,
+                   Arguments<(ins LLVM_Type:$a, LLVM_Type:$b)>,
+                   Results<(outs LLVM_Type:$res)> {
+  let llvmBuilder = [{
+    llvm::Module *module = builder.GetInsertBlock()->getModule();
+    llvm::Function *fn = llvm::Intrinsic::getDeclaration(
+        module, llvm::Intrinsic::copysign, {$a->getType(), $b->getType()});
+    $res = builder.CreateCall(fn, {$a, $b});
+  }];
+}
+
+def LLVM_TanhOp : LLVM_Op<"tanh", [NoSideEffect]>,
```

Comment:
If there is no `tanh` op in LLVM, it should not be an LLVM Op. Create it in some dialect and implement a lowering from that dialect to the LLVM dialect instead, so the translation from the LLVM dialect to LLVM IR is as simple as possible. Having magic constants in the translation wouldn't cut it.

If you implement your non-trivial conversion in as a conversion pattern, it is significantly easier to replace.